dask/array/core.py

```diff
@@ -3052,7 +3052,7 @@ def insert_to_ooc(arr, out, lock=True, region=None,
     if region:
         slices = [fuse_slice(region, slc) for slc in slices]
 
-    name = 'store-%s' % arr.name
+    name = 'store-%s' % arr.name + tokenize(region)
```

Comment:
In other operations we tend to include the input in tokenize rather than extend the name.  Otherwise names can grow fairly long.